xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java

```diff
@@ -17,12 +18,13 @@ public ClientChannelConfiguration channelConfig(ChannelHandlerContext ctx) {
     return ChannelConfiguration.clientConfig(ctx.channel().eventLoop());
   }
 
-  public abstract Client createClient(ChannelHandlerContext ctx, ClientConfig config);
+  public abstract Client createClient(
+      ChannelHandlerContext ctx, ClientConfig config, XioTracing tracing);
 
-  public Client getClient(ChannelHandlerContext ctx, ClientConfig config) {
+  public Client getClient(ChannelHandlerContext ctx, ClientConfig config, XioTracing tracing) {
     Client client = ctx.channel().attr(CLIENT_KEY).get();
     if (client == null) {
-      client = createClient(ctx, config);
+      client = createClient(ctx, config, tracing);
       ctx.channel().attr(CLIENT_KEY).set(client);
```

Comment:
Can we put `XioTracing` into the `ClientFactory` constructor and store it as state on this class? If we need to do dynamic changes to the location of the zipkin server, let's hide that behind `XioTracing` instead of passing a new `XioTracing` into `getClient`.

Reply:
Ok that works for me.

Reply:
ok now ClientFactory(XioTracing tracing) is a thing.  when you call getClient you don't need to have a tracing (yay!)

Reply:
fixed